### PR TITLE
feat: implement hideTableHeader prop in table component

### PR DESCRIPTION
**Pull request by lyon93**
fix: #1793

## Changes proposed in this PR:
-

- [ ] I have followed (at least) the [PR section of the contributing guide](https://github.com/nexxtway/react-rainbow/blob/master/CONTRIBUTING.md#submitting-a-pull-request).

@nexxtway/react-rainbow

### CODE REVIEW

**Review by yvmunayev — COMMENTED** [review]
I think a more optimal solution would be to use <RenderIf> at the <thead> level

**Comment by LeandroTorresSicilia** [comment]
@lyon93 check here there are conflicts

**Comment by TahimiLeonBravo** [comment]
@lyon93 please check the conflicts that you have, need be resolved

File: src/components/Table/head/__test__/head.spec.js [hunk 1]
```diff
@@ -22,6 +22,16 @@ describe('<Head />', () => {
...
+    it('should not show the header columns', () => {
+        const component = mount(<Head columns={columns} hideTableHeader />);
+        const header = component.find('Header');
+        expect(header.at(0).prop('hideTableHeader')).toBeTruthy();
+    });
+    it('should show the header columns', () => {
+        const component = mount(<Head columns={columns} />);
+        const header = component.find('Header');
+        expect(header.at(0).prop('hideTableHeader')).toBeFalsy();
+    });
```
Comment (reiniergs) [inline src/components/Table/head/__test__/head.spec.js:34]: these tests ^^ should check the header is not visible, checking the prop value when is being passed is testing React :) .. not the component behavior.

Although, the implementation of the feature looks good to me

**Pull request closed without merging**